src/util/newconfig/ArrayView.hpp

```diff
@@ -66,45 +184,20 @@ class ArrayView {
      * @param idx Index of the value to retrieve.
      * @return ValueView at the specified index.
      */
-    ValueView
+    [[nodiscard]] ValueView
     valueAt(std::size_t idx) const;
 
     /**
      * @brief Returns the number of elements in the array.
      *
      * @return Number of elements in the array.
-     * @throw std::logic_error if somehow ArrayView doesn't exist
      */
-    size_t
+    [[nodiscard]] size_t
     size() const;
 
-    /**
-     * @brief Returns an iterator to the beginning of the values.
-     *
-     * @return Iterator to the beginning of the values.
-     */
-    auto
-    beginValues() const
-    {
-        ASSERT(clioConfig_.map_.contains(prefix_), "Current string is prefix, not full key");
-        return std::get<Array>(clioConfig_.map_.at(prefix_)).begin();
-    }
-
-    /**
-     * @brief Returns an iterator to the end of the values.
-     *
-     * @return Iterator to the end of the values.
-     */
-    auto
-    endValues() const
-    {
-        ASSERT(clioConfig_.map_.contains(prefix_), "Current string is prefix, not full key");
-        return std::get<Array>(clioConfig_.map_.at(prefix_)).end();
-    }
-
 private:
     std::string const prefix_;
```

Comment:
We should stop using const on member variables.. i know it's a bit controversial but the reasoning is the same as for reference member variables (more or less). 

We should probably also enable this check: https://clang.llvm.org/extra/clang-tidy/checks/cppcoreguidelines/avoid-const-or-ref-data-members.html